tensorboard/webapp/metrics/views/main_view/common_selectors.ts

```diff
@@ -287,10 +289,49 @@ export const getPotentialHparamColumns = createSelector(
       sortable: true,
       movable: true,
       filterable: true,
+      hidable: true,
     }));
   }
 );
 
+export const getSelectableColumns = createSelector(
+  getPotentialHparamColumns,
+  getDashboardDisplayedHparamColumns,
+  (potentialColumns, currentColumns) => {
+    const currentColumnNames = new Set(currentColumns.map(({name}) => name));
+    return potentialColumns.filter((columnHeader) => {
+      return !currentColumnNames.has(columnHeader.name);
+    });
+  }
+);
+
+/** Returns a list of columns that have been sorted into logical groups.
+ *
+ * Column order: | RUN | experimentAlias | HPARAMs | other |
+ */
+export const getGroupedColumns = (
+  headersSelector: Selector<object, ColumnHeader[]>
+) =>
+  createSelector(
+    headersSelector,
+    getDashboardDisplayedHparamColumns,
+    (tableHeaders, hparamHeaders): ColumnHeader[] => {
+      return [
+        ...tableHeaders.filter((header) => header.type === 'RUN'),
+        ...tableHeaders.filter(
+          (header) =>
+            header.type === 'CUSTOM' && header.name === 'experimentAlias'
+        ),
+        ...hparamHeaders,
+        ...tableHeaders.filter(
+          (header) =>
+            header.type !== 'RUN' &&
+            !(header.type === 'CUSTOM' && header.name === 'experimentAlias')
+        ),
+      ];
```

Comment:
I'm not sure this approach of running a series of filters, then dumping everything which doesn't match at the end will scale if you end up with a handful of additional confitions.

```suggestion
      const headersGroups: Record<string, ColumnHeader[]> = {
        RUN: [],
        EXPERIMENT_ALIAS: [],
        OTHER: [],
      };
      tableHeaders.forEach((header) => {
        if (header.type === 'RUN') {
            headersGroups['RUN'].push(header);
            return;
        }
        if (header.type === 'CUSTOM' && header.name === 'experimentAlias') {
            headerGroups['EXPERIMENT_ALIAS'].push(header);
            return;
        }
        headersGroups['OTHER'].push(header);
      })
      return [
          ...headerGroups['RUN'],
          ...headerGroups['EXPERIMENT_ALIAS'],
          ...hparamHeaders,
          ...headerGroups['OTHER'],
```

Reply:
I was also wondering about a more robust way to do this and you gave me an idea. I implemented something along these lines in `/widgets/data_table/utils.ts:groupColumns`, PTAL!